packages/loot-core/src/server/api.js

```diff
@@ -160,6 +161,35 @@ handlers['api/load-budget'] = async function ({ id }) {
   }
 };
 
+handlers['api/download-budget'] = async function ({ syncId, password }) {
```

Comment:
I think in this function we want to check if the file is already downloaded, and if so do nothing right?

Or change the API the be `api.loadBudget` and it would magically automatically download it if it's not already downloaded. What do you think?

Reply:
> I think in this function we want to check if the file is already downloaded, and if so do nothing right?

I assumed `cloudStorage.download` already did that because `download-budget` checks for the `file-exists` error, but it seems like it doesn’t. I’ll add that!

> Or change the API the be `api.loadBudget` and it would magically automatically download it if it's not already downloaded. What do you think?

Currently you have to use the sync id to download, and the budget id to open. (The budget id is helpfully returned from the `handlers['download-budget']` API, once the download completes).

Reply:
(Fortunately, `get-budgets` returns sync file IDs for local budgets, so I’m able to fetch them that way).

Updated the code to look for a local budget and trigger a full sync on it instead of re-downloading if possible. I think this is ready for another review!